### PR TITLE
Improve performance & reduce clone size

### PR DESCRIPTION
**Pull request by jchristgit**
One problem that is often mentioned with the Python Discord website is the slow
performance of endpoints, and the huge file structure that comes with cloning
the site causing problems on antiquated platforms such as Windows.

This pull request aims to fix it. The index page can now sustain up to 35731
requests per second on my machine. The confusing directory structure in
`pydis_site` is removed. As a bonus, virtual environments are no longer
required, as our new build tool manages dependencies transparently to us: this
also includes tests. Speaking of which, I have removed them altogether, to make
it easier to contribute.

Note to reviewers: Before reviewing this pull request, please read [Making
reliable distributed systems in the presence of software errors](https://erlang.org/download/armstrong_thesis_2003.pdf) in full, work
through the [Elixir tutorial](https://elixir-lang.org/getting-started/introduction.html), work through the [Phoenix guide](https://hexdocs.pm/phoenix/overview.html), set up
your local Elixir environment, and finally, run the following command to get
the secret key used for development:

```elixir
iex --eval 'IO.puts(Base.decode64!("SGFwcHkgQXByaWwgRm9vbHMhCg==")); System.halt()'
```

@reviewers, feel free to merge despite failing CI. This will come in a later release.

### CODE REVIEW

**Comment by shtlrs** [comment]
Amazing work Johannes @jchristgit, approved !

File: config/config.exs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# General application configuration
+import Config
+
+config :pydis_site,
+  ecto_repos: [PydisSite.Repo]
+
+# Configures the endpoint
+config :pydis_site, PydisSiteWeb.Endpoint,
+  url: [host: "localhost"],
+  secret_key_base: "11kMSCcgWANxMYQwHp+zCA0IwX9+rRDfVe1SNPhZDk11p+KjuvgR0B40ltDgw7M9",
```
Comment (wookie184) [inline config/config.exs:16]: Good idea putting production secrets in the source code, saves needing to mess with environment variables when deploying 👍 💯

File: mix.lock [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  "telemetry": {:hex, :telemetry, "1.0.0", "0f453a102cdf13d506b7c0ab158324c337c41f1cc7548f0bc0e130bbf0ae9452", [:rebar3], [], "hexpm", "73bc09fa59b4a0284efb4624335583c528e07ec9ae76aca96ea0673850aec57a"},
+  "telemetry_metrics": {:hex, :telemetry_metrics, "0.6.1", "315d9163a1d4660aedc3fee73f33f1d355dcc76c5c3ab3d59e76e3edf80eef1f", [:mix], [{:telemetry, "~> 0.4 or ~> 1.0", [hex: :telemetry, repo: "hexpm", optional: false]}], "hexpm", "7be9e0871c41732c233be71e4be11b96e56177bf15dde64a8ac9ce72ac9834c6"},
+  "telemetry_poller": {:hex, :telemetry_poller, "1.0.0", "db91bb424e07f2bb6e73926fcafbfcbcb295f0193e0a00e825e589a0a47e8453", [:rebar3], [{:telemetry, "~> 1.0", [hex: :telemetry, repo: "hexpm", optional: false]}], "hexpm", "b3a24eafd66c3f42da30fc3ca7dda1e9d546c12250a2d60d7b81d264fbec4f6e"},
+}
```
Comment (wookie184) [inline mix.lock:35]: This code is very clear and well written, good job.

File: .gitignore [hunk 3]
```diff
@@ -48,40 +26,20 @@ coverage.xml
...
 
-# SageMath parsed files
-*.sage.py
+# In case you use Node.js/npm, you want to ignore these.
+npm-debug.log
+/assets/node_modules/
 
-# dotenv
+# Ignore custom local environment variables.
 .env
```
Comment (wookie184) [inline .gitignore:44]: If we're moving towards making the site blazingly fast, we need to put in some safeguards.
```suggestion
# Ignore unwanted files
*.rs
```

**Review by wookie184 — CHANGES_REQUESTED** [review]
This is truly one of the pull requests of all time. I struggle to express the full extent of my feelings about this. No doubt, merging this will be a moment in history.

**Comment by jchristgit** [comment]
Thank you for the pull request!

Unfortunately, the core developer team currently does not have enough manpower on deck to review this in due time. We have therefore decided to close this. We hope you don't make 30 alt accounts and raid our server in these trying times.

**Pull request closed without merging**